.github/workflows/common-api-ci.yml

```diff
@@ -0,0 +1,198 @@
+
+        # If a particular API project needs to do some special docker setup
+        # (e.g. Sitewide Search sets up synonyms for Elasticsearch), it should
+        # be encapsulated in a docker-setup.sh script.
+      - name: project-specific Docker setup.
+        shell: bash
+        run: |
+            if [[ -e 'integration-tests/bin/docker-setup.sh' ]]; then
+              CONTAINER_ID=$(docker ps -q --filter "ancestor=elasticsearch:7.17.5")
+              integration-tests/bin/docker-setup.sh $CONTAINER_ID
+            fi
+
+      - name: Download Published Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.artifact-name }}-api-ubuntu-latest
+          path: built-api
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          global-json-file: global.json
+          source-url: "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json"
+
+      - name: Load Data into elasticsearch & prepare for tests
+        env:
+          ELASTIC_SEARCH_HOST: http://localhost:${{ job.services.elasticsearch.ports[9200] }}
+        shell: bash
+        run: |
+              ## Create test output and API logging location
+              mkdir -p integration-tests/target
+
+              ## Load the elasticsearch data
+              ./integration-tests/bin/load-integration-data.sh
+
+      - name: Start API
+        env:
+          Elasticsearch__Servers: http://localhost:${{ job.services.elasticsearch.ports[9200] }}
+          ASPNETCORE_LOGGING__CONSOLE__DISABLECOLORS: true
+          API_URL: http://localhost:5000
+          SLEEP_TIMEOUT: 5
+          WAIT_DURATION: 120
+          APP_PATH: ${{ github.workspace }}/built-api/${{ inputs.artifact-name }}-api
+        shell: bash
+        run: |
+              ## Start the app and log output
+              ## NOTE: We must change directory because in order for dotnet to be able to locate the appsettings.json file.
+              cd $APP_PATH && ${{ github.workspace }}/integration-tests/bin/start-api.sh > ${{ github.workspace }}/integration-tests/target/api_log.txt 2>&1 &
+
+              time_waited=1
+              echo "Checking status of ${API_URL}."
+              until $(curl --output /dev/null --silent --fail ${API_URL}); do
+                echo $?
+
+                if [ $time_waited -ge $WAIT_DURATION ]; then
+                    echo "Waited past duration. Exiting"
+                    cat integration-tests/target/api_log.txt
+                    exit 1
+                fi
+
+                sleep $SLEEP_TIMEOUT
+                time_waited=$((time_waited + SLEEP_TIMEOUT))
+              done
+
+              echo "API is up"
+
+      - name: Run Integration Test
+        id: integration_tests
+        shell: bash
+        run: |
+              ## Run Karate
+              cd integration-tests && ./bin/karate ./features
+
+      - name: Upload Integration test results
+        if: always()
```

Comment:
Is this the correct syntax here? The docs say it should be `if: {{ always() }}`
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#always

Reply:
The curly brackets are optional on the `if` ([Docs](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idif))

> When you use expressions in an if conditional, you can, optionally, omit the ${{ }} expression syntax because GitHub Actions automatically evaluates the if conditional as an expression. However, this exception does not apply everywhere.